Lib/shutil.py

```diff
@@ -478,12 +478,12 @@ def _copytree(entries, src, dst, symlinks, ignore, copy_function,
                         continue
                     # otherwise let the copy occur. copy2 will raise an error
                     if srcentry.is_dir():
-                        copytree(srcobj, dstname, symlinks, ignore,
+                        copytree(srcname, dstname, symlinks, ignore,
```

Comment:
This removes the speedup originally introduced. I think you can just do this:

```
ignored_names = ignore(os.fspath(src), list({x.path for x in entries}))
```

There should be a unit test which makes sure the callback function receives a string regardless of the type passed as input (str, pathlib.Path, os.DirEntry).

Reply:
I have made the requested changes; please review again.

Reply:
Sorry, I missed something for the test.

Reply:
Ok, I think it should be good now.